omod/src/main/webapp/resources/scripts/conditionlist/controllers/addcondition.controller.js

```diff
@@ -31,25 +35,46 @@ function ConditionController($scope, RestfulService, ConditionModel, ConceptMode
                 emr.errorAlert("Error Saving condition");
             });
         }
+    }
 
     self.initCondition = self.initCondition || function () {
             $scope.patientUuid = CommonFunctions.extractUrlArgs(window.location.search)['patientId'];
             $scope.condition = new ConditionModel($scope.patientUuid);
+            self.showEndDate();
         }
-
+        
     self.validateCondition = self.validateCondition || function () {
-            var concept = $scope.concept.concept;
-            if (concept !== null) {
-                $scope.condition.concept = new ConceptModel(concept.uuid, concept.display);
-            } else {
-                // noncoded diagnosis
-                $scope.condition.conditionNonCoded = "NON_CODED:" + $scope.concept.word;
-            }
-
-            $scope.condition.onSetDate = self.getSelectedDate();
+        var concept ;
+        if($scope.concept.concept.uuid)
+        {
+            concept=$scope.concept.concept;
+        }
+        else{
+            concept=$scope.condition.concept;
+        }
+        if (concept !== null) {
+            $scope.condition.concept = new ConceptModel(concept.uuid, concept.display);
+        } else {
+            // noncoded diagnosis
+            $scope.condition.conditionNonCoded = "NON_CODED:" + $scope.concept.word;
+        }
+        $scope.condition.onSetDate = self.getSelectedDate();
+        if($scope.condition.status=='INACTIVE')
+        {
+            $scope.condition.endDate = self.getEndDate();
+        }
+        self.saveCondition();
+    }
 
-            self.saveCondition();
+    self.showEndDate = self.showEndDate || function() {
+        var groups = document.getElementsByClassName("group");
+        if($scope.condition.status=='INACTIVE'){
```

Comment:
Why do we repeat 'INACTIVE'?

Reply:
Here I'm checking if condition status is 'INACTIVE', in order to make the end date field visible only when we choose the inactive radio button.